src/scheme.mli

```diff
@@ -0,0 +1,23 @@
+(** A collection of rules for one or multiple directories. *)
+
+open! Stdune
+
+(** Generic representation of a scheme *)
+module Gen : sig
+  type 'rules t =
+    | Empty
+    | Union of 'rules t * 'rules t
+    | Approximation of Dir_set.t * 'rules t
+    | Finite of 'rules Path.Build.Map.t
+    | Thunk of (unit -> 'rules t)
+end
+
+module type S = Scheme_intf.S with module Gen := Gen
+
+module Make(Directory_rules : sig
```

Comment:
This API feels a bit odd to me. I.e. I don't really know how I would call `Directory_rules`; it is a module that can only produce empty values. Looking at the implementation, we only need `union` in `evaluate` and `empty` in `rules_here`. What do you think about only exposing the polymorphic API, passing `union` to `evaluate` and replacing `'rules` by `'rules option` in `rules_here`?

Reply:
I like that. I'm making the change.